classes/services/queryBuilders/SubmissionListQueryBuilder.inc.php

```diff
@@ -44,6 +44,38 @@ public function filterBySections($sectionIds) {
 	 * @return object Query object
 	 */
 	public function appGet($q) {
```

Comment:
Thinking broadly about the division of SQL between the DAOs and the service layer classes, what is the end-game for the DAOs? For example, would they retain insert/delete/fetch, leaving the more business-logic-intensive joins to service classes? (That's as much for @kaschioudi as anyone)

Reply:
@kaschioudi, pinging you on the above comment!

Reply:
Thanks Alec for the reminder. I totally forgot about this. So here's my point of view.

Generally services are helper objects which handle complex business logic usually by controlling multiple domain objects. For OJS that means Services should orchestrate interactions between group of DAO and/or query builders. CRUD and simple operations remain on DAOs and complex queries go into entity query builders. Once we replace ADODB (https://github.com/pkp/pkp-lib/issues/2493), we will probably introduce model (ActiveRecord implementation) objects which will centralize CRUD and complex queries under a single class.

Reply:
Thanks, that makes sense -- and provides a good place for cross-object queries to live, which we've never had before.